doc/source/ray-core/type-hint.rst

```diff
@@ -0,0 +1,121 @@
+Type Hints in Ray
+==================
+
+Ray provides comprehensive support for Python type hints with both remote functions and actors.
+This enables better IDE support, static type checking, and improved code maintainability in distributed Ray applications.
+
+Overview
+--------
+
+In most cases, Ray applications can use type hints without any modifications to existing code.
+Ray automatically handles type inference for standard remote functions and basic actor usage patterns.
+
+However, certain patterns require specific approaches to ensure proper type annotation,
+particularly when working with actor references.
+
+Remote Functions
+----------------
+
+Ray remote functions support standard Python type annotations without additional configuration.
+The ``@ray.remote`` decorator preserves the original function signature and type information.
+
+.. code-block:: python
+
+    import ray
+
+    @ray.remote
+    def add_numbers(x: int, y: int) -> int:
+        return x + y
+
+    # Type hints work seamlessly with remote function calls
+    result_ref = add_numbers.remote(5, 3)
+    result: int = ray.get(result_ref)
```

Comment:
The `int` here doesn't need to be explicitly declared here and will get inferred, right? (I just tried this) -- maybe remove the `int` and instead add a comment above that states that the `result` variable is inferred as `int` :)